sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -137,8 +137,7 @@ class CodeGenContext {
     dataType match {
       case _ if isPrimitiveType(jt) => s"$row.set${primitiveTypeName(jt)}($ordinal, $value)"
       case t: DecimalType => s"$row.setDecimal($ordinal, $value, ${t.precision})"
-      // The UTF8String may came from UnsafeRow, otherwise clone is cheap (re-use the bytes)
-      case StringType => s"$row.update($ordinal, $value.clone())"
+      case StringType => s"$row.update($ordinal, $value)"
```

Comment:
SafeProjection may be not the only one projection that turn an UnsafeRow into InternalRow. For some operators that accept both safe row and unsafe row, that will use generated mutable projection, it will got crupted UTF8String if we remove the copy here.